### ISSUE TITLE
W: GPG error: http://apt.datadoghq.com beta Release

### ISSUE THREAD

**Issue by matthewbennink** [DataDog/heroku-buildpack-datadog#32]
Noticed this error as part of the slug compile. It lead me to [this page|https://github.com/DataDog/dd-agent/wiki/Debian-Ubuntu-package:-Expired-Signature], which suggests it's possible to update the key prior to downloading / installing the package. Is this something that should/could be added to the buildpack? Or perhaps it will go away when it's no longer using the beta channel?

Full messages:
```
W: GPG error: http://apt.datadoghq.com beta Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 226AE980C7A7DA52
...
WARNING: The following packages cannot be authenticated!
  datadog-agent
```

**Comment by jyee** [comment]
Thanks, I've had this issue in my todo list for a while. (see the [outstanding issues in the changelog](https://github.com/DataDog/heroku-buildpack-datadog/blob/master/CHANGELOG.md#unreleased--outstanding-issues)).

At the moment it's a minor annoyance, since the package will still download and install, but I do need to address the key acceptance issue as Datadog will rotate keys soon: https://help.datadoghq.com/hc/en-us/articles/360000886852-Agent-APT-Repo-Key-Rotation

**Comment by mottalrd** [comment]
Hello, we got this message too today, how do I know if we need to take any action?
Thank you

**Comment by arapulido** [comment]
I have been trying to fix this issue, but it seems to be working very differently in Ubuntu 16.04 and Ubuntu 18.04.

In Ubuntu 18.04 the issue was a wrong format in the keyring, which I have fixed in https://github.com/DataDog/heroku-buildpack-datadog/tree/ara.pulido/do-not-bypass-gpg

But in Ubuntu 16.04, it seems that no matter what I do, `apt-get update` completely ignores `-o Dir::Etc::Trusted` option when passing it to `apt-key` and, therefore, it never gets read. The same keyring copied to `/etc/apt/trusted.gpg.d` works fine.

I can prepare a PR based on my branch, as it at least fixes the issue on 18.04 (and leaves 16.04 as it is), or we can continue ignoring this. @jyee what do you think?
